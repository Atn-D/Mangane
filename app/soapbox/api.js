'use strict';

import axios from 'axios';
import LinkHeader from 'http-link-header';

export const getLinks = response => {
  const value = response.headers.link;

  if (!value) {
    return { refs: [] };
  }

  return LinkHeader.parse(value);
};

const getToken = (getState, authType) =>
  getState().getIn(['auth', authType, 'access_token']);

export default (getState, authType = 'user') => {
  const accessToken = getToken(getState, authType);

  return axios.create({
    headers: Object.assign(accessToken ? {
      'Authorization': `Bearer ${accessToken}`,
    } : {}),

    transformResponse: [function(data) {
      try {
        return JSON.parse(data);
      } catch(Exception) {
        return data;
      }
    }],
  });
};
