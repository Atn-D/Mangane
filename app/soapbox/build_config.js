// @preval
/**
 * Build config: configuration set at build time.
 * @module soapbox/build_config
 */

const { trim } = require('lodash');

const {
  BACKEND_URL,
  FE_BASE_PATH,
} = process.env;

const sanitizeURL = url => {
  try {
    return new URL(url).toString();
  } catch {
    return '';
  }
};

// Run Soapbox FE from a subdirectory.
const getFeBasePath = () => {
  return `/${trim(FE_BASE_PATH, '/')}`;
};

// JSON.parse/stringify is to emulate what @preval is doing and avoid any
// inconsistent behavior in dev mode
const sanitize = obj => JSON.parse(JSON.stringify(obj));

module.exports = sanitize({
  BACKEND_URL: sanitizeURL(BACKEND_URL),
  FE_BASE_PATH: getFeBasePath(),
});
