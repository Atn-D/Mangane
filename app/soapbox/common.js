'use strict';

import Rails from 'rails-ujs';

export function start() {
  require('fork-awesome/css/fork-awesome.css');
  require('@fonticonpicker/react-fonticonpicker/dist/fonticonpicker.base-theme.react.css');
  require('@fonticonpicker/react-fonticonpicker/dist/fonticonpicker.material-theme.react.css');
  require.context('../images/', true);

  try {
    Rails.start();
  } catch (e) {
    // If called twice
  }
};
