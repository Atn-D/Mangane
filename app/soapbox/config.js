export default {
  features: {
    patron: true,
  },
}
