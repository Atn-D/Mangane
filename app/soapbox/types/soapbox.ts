import {
  PromoPanelItemRecord,
  FooterItemRecord,
  CryptoAddressRecord,
  SoapboxConfigRecord,
} from 'soapbox/normalizers/soapbox/soapbox_config';

type PromoPanelItem = ReturnType<typeof PromoPanelItemRecord>;
type FooterItem = ReturnType<typeof FooterItemRecord>;
type CryptoAddress = ReturnType<typeof CryptoAddressRecord>;
type SoapboxConfig = ReturnType<typeof SoapboxConfigRecord>;

export {
  PromoPanelItem,
  FooterItem,
  CryptoAddress,
  SoapboxConfig,
};
