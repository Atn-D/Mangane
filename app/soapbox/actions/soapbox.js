import api from '../api';

export const SOAPBOX_CONFIG_REQUEST_SUCCESS = 'SOAPBOX_CONFIG_REQUEST_SUCCESS';
export const SOAPBOX_CONFIG_REQUEST_FAIL    = 'SOAPBOX_CONFIG_REQUEST_FAIL';

export const SOAPBOX_PATCH_REQUEST = 'SOAPBOX_PATCH_REQUEST';
export const SOAPBOX_PATCH_SUCCESS = 'SOAPBOX_PATCH_SUCCESS';
export const SOAPBOX_PATCH_FAIL    = 'SOAPBOX_PATCH_FAIL';

export function fetchSoapboxConfig() {
  return (dispatch, getState) => {
    api(getState).get('/api/pleroma/frontend_configurations').then(response => {
      if (response.data.soapbox_fe) {
        dispatch(importSoapboxConfig(response.data.soapbox_fe));
      } else {
        api(getState).get('/instance/soapbox.json').then(response => {
          dispatch(importSoapboxConfig(response.data));
        }).catch(error => {
          dispatch(soapboxConfigFail(error));
        });
      }
    }).catch(error => {
      api(getState).get('/instance/soapbox.json').then(response => {
        dispatch(importSoapboxConfig(response.data));
      }).catch(error => {
        dispatch(soapboxConfigFail(error));
      });
    });
  };
}

export function importSoapboxConfig(soapboxConfig) {
  return {
    type: SOAPBOX_CONFIG_REQUEST_SUCCESS,
    soapboxConfig,
  };
}

export function soapboxConfigFail(error) {
  if (!error.response) {
    console.error('Unable to obtain soapbox configuration: ' + error);
  }
  return {
    type: SOAPBOX_CONFIG_REQUEST_FAIL,
    error,
    skipAlert: true,
  };
}

export function patchSoapbox(params) {
  console.log(params);
  return (dispatch, getState) => {
    dispatch(patchSoapboxRequest());
    return api(getState)
      .patch('/api/pleroma/admin/config', params)
      .then(response => {
        dispatch(patchSoapboxSuccess(response.data));
      }).catch(error => {
        dispatch(patchSoapboxFail(error));
      });
  };
}

export function patchSoapboxRequest() {
  return {
    type: SOAPBOX_PATCH_REQUEST,
  };
}

export function patchSoapboxSuccess(me) {
  return (dispatch, getState) => {
    dispatch({
      type: SOAPBOX_PATCH_SUCCESS,
      me,
    });
  };
}

export function patchSoapboxFail(error) {
  return {
    type: SOAPBOX_PATCH_FAIL,
    error,
  };
};
