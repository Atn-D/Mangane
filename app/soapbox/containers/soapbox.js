'use strict';

import React from 'react';
import { Provider, connect } from 'react-redux';
import PropTypes from 'prop-types';
import SoapboxPropTypes from 'soapbox/utils/soapbox_prop_types';
import Helmet from 'soapbox/components/helmet';
import classNames from 'classnames';
import configureStore from '../store/configureStore';
import { INTRODUCTION_VERSION } from '../actions/onboarding';
import { Switch, BrowserRouter, Route } from 'react-router-dom';
import { ScrollContext } from 'react-router-scroll-4';
import UI from '../features/ui';
// import Introduction from '../features/introduction';
import { fetchCustomEmojis } from '../actions/custom_emojis';
import { hydrateStore } from '../actions/store';
import { IntlProvider, addLocaleData } from 'react-intl';
import { getLocale } from '../locales';
import initialState from '../initial_state';
import ErrorBoundary from '../components/error_boundary';
import { fetchInstance } from 'soapbox/actions/instance';
import { fetchSoapboxConfig } from 'soapbox/actions/soapbox';
import { fetchMe } from 'soapbox/actions/me';
import PublicLayout from 'soapbox/features/public_layout';
import { getSettings } from 'soapbox/actions/settings';
import { themeDataToCss } from 'soapbox/utils/theme';
import { generateTheme } from 'soapbox/actions/theme';

export const store = configureStore();
const hydrateAction = hydrateStore(initialState);

store.dispatch(hydrateAction);
store.dispatch(fetchMe());
store.dispatch(fetchInstance());
store.dispatch(fetchSoapboxConfig());
store.dispatch(fetchCustomEmojis());

const mapStateToProps = (state) => {
  const me = state.get('me');
  const account = state.getIn(['accounts', me]);
  const showIntroduction = account ? state.getIn(['settings', 'introductionVersion'], 0) < INTRODUCTION_VERSION : false;
  const settings = getSettings(state);

  return {
    showIntroduction,
    me,
    reduceMotion: settings.get('reduceMotion'),
    systemFont: settings.get('systemFont'),
    dyslexicFont: settings.get('dyslexicFont'),
    demetricator: settings.get('demetricator'),
    locale: settings.get('locale'),
    themeCss: themeDataToCss(state.get('theme')),
    themeMode: settings.get('themeMode'),
    brandColor: state.getIn(['soapbox', 'brandColor']),
  };
};

@connect(mapStateToProps)
class SoapboxMount extends React.PureComponent {

  static propTypes = {
    showIntroduction: PropTypes.bool,
    me: SoapboxPropTypes.me,
    reduceMotion: PropTypes.bool,
    systemFont: PropTypes.bool,
    dyslexicFont: PropTypes.bool,
    demetricator: PropTypes.bool,
    locale: PropTypes.string.isRequired,
    themeCss: PropTypes.string,
    brandColor: PropTypes.string,
    themeMode: PropTypes.string,
    dispatch: PropTypes.func,
  };

  maybeUpdateTheme = prevProps => {
    const updates = [
      this.props.brandColor !== prevProps.brandColor,
      this.props.themeMode  !== prevProps.themeMode,
    ];
    if (updates.some(u => u)) this.props.dispatch(
      generateTheme(this.props.brandColor, this.props.themeMode)
    );
  }

  componentDidUpdate(prevProps) {
    this.maybeUpdateTheme(prevProps);
  }

  render() {
    const { me, themeCss, reduceMotion, systemFont, dyslexicFont, demetricator, locale } = this.props;
    if (me === null) return null;

    const { localeData, messages } = getLocale();
    addLocaleData(localeData);

    // Disabling introduction for launch
    // const { showIntroduction } = this.props;
    //
    // if (showIntroduction) {
    //   return <Introduction />;
    // }

    const bodyClass = classNames('app-body', {
      'system-font': systemFont,
      'no-reduce-motion': !reduceMotion,
      'dyslexic': dyslexicFont,
      'demetricator': demetricator,
    });

    return (
      <IntlProvider locale={locale} messages={messages}>
        <>
          <Helmet>
            <body className={bodyClass} />
            {themeCss && <style id='theme' type='text/css'>{`:root{${themeCss}}`}</style>}
          </Helmet>
          <BrowserRouter>
            <ScrollContext>
              <Switch>
                {!me && <Route exact path='/' component={PublicLayout} />}
                <Route exact path='/about/:slug?' component={PublicLayout} />
                <Route path='/' component={UI} />
              </Switch>
            </ScrollContext>
          </BrowserRouter>
        </>
      </IntlProvider>
    );
  }

}

export default class Soapbox extends React.PureComponent {

  render() {
    return (
      <Provider store={store}>
        <ErrorBoundary>
          <SoapboxMount />
        </ErrorBoundary>
      </Provider>
    );
  }

}
