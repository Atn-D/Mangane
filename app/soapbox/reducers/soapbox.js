import { SOAPBOX_CONFIG_IMPORT } from '../actions/soapbox';
import { Map as ImmutableMap, fromJS } from 'immutable';

const initialState = ImmutableMap();

export default function soapbox(state = initialState, action) {
  switch(action.type) {
  case SOAPBOX_CONFIG_IMPORT:
    return ImmutableMap(fromJS(action.soapboxConfig));
  default:
    return state;
  }
};
