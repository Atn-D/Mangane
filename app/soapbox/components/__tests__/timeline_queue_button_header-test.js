import React from 'react';
import TimelineQueueButtonHeader from '../timeline_queue_button_header';
import { createComponentWithIntl } from 'soapbox/test_helpers';
import { defineMessages } from 'react-intl';

const messages = defineMessages({
  queue: { id: 'status_list.queue_label', defaultMessage: 'Click to see {count} new {count, plural, one {post} other {posts}}' },
});

describe('<TimelineQueueButtonHeader />', () => {
  it('renders correctly', () => {
    expect(createComponentWithIntl(
      <TimelineQueueButtonHeader
        key='timeline-queue-button-header'
        onClick={() => {}} // eslint-disable-line react/jsx-no-bind
        count={0}
        message={messages.queue}
      />
    ).toJSON()).toMatchSnapshot();

    expect(createComponentWithIntl(
      <TimelineQueueButtonHeader
        key='timeline-queue-button-header'
        onClick={() => {}} // eslint-disable-line react/jsx-no-bind
        count={1}
        message={messages.queue}
      />
    ).toJSON()).toMatchSnapshot();

    expect(createComponentWithIntl(
      <TimelineQueueButtonHeader
        key='timeline-queue-button-header'
        onClick={() => {}} // eslint-disable-line react/jsx-no-bind
        count={9999999}
        message={messages.queue}
      />
    ).toJSON()).toMatchSnapshot();
  });
});
