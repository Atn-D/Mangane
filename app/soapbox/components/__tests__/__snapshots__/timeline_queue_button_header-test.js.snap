// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimelineQueueButtonHeader /> renders correctly 1`] = `
<div
  className="timeline-queue-header hidden"
>
  <a
    className="timeline-queue-header__btn"
    onClick={[Function]}
  >
    <div
      className="svg-icon"
    >
      <svg
        id={
          Object {
            "process": [Function],
          }
        }
      />
    </div>
  </a>
</div>
`;

exports[`<TimelineQueueButtonHeader /> renders correctly 2`] = `
<div
  className="timeline-queue-header"
>
  <a
    className="timeline-queue-header__btn"
    onClick={[Function]}
  >
    <div
      className="svg-icon"
    >
      <svg
        id={
          Object {
            "process": [Function],
          }
        }
      />
    </div>
    Click to see 1 new post
  </a>
</div>
`;

exports[`<TimelineQueueButtonHeader /> renders correctly 3`] = `
<div
  className="timeline-queue-header"
>
  <a
    className="timeline-queue-header__btn"
    onClick={[Function]}
  >
    <div
      className="svg-icon"
    >
      <svg
        id={
          Object {
            "process": [Function],
          }
        }
      />
    </div>
    Click to see 9999999 new posts
  </a>
</div>
`;
